File: 982-MinimumIncrementToMakeArrayUnique/982-MinimumIncrementToMakeArrayUnique.cpp
// Last updated: 06/07/2026, 19:06:51
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        cin.tie(NULL);
        cout.tie(NULL);
        sort(nums.begin(), nums.end());
        int result = 0;
        int prev = -1;
        for (int& num : nums){
            if (num <= prev) {
                result += prev - num + 1;
                num += prev - num + 1;
            }
            prev = num;
        }
        return result;
    }
};

File: 982-MinimumIncrementToMakeArrayUnique/982-MinimumIncrementToMakeArrayUnique.py
# Last updated: 06/07/2026, 19:06:57
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        nums.sort()
        result = 0
        prev = -1
        for i in range(len(nums)):
            cur = nums[i]
            if (cur <= prev):
                result += prev - cur + 1
                cur += prev - cur + 1
            prev = cur

        return result